docs/understanding-spark-bench.md

```diff
@@ -0,0 +1,248 @@
+
+# Understanding `spark-bench`
+
+`spark-bench` is a flexible system for benchmarking and simulating Spark jobs. 
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Data Generation](#data-generation)
+- [Workloads](#workloads)
+  - [Types of Workloads](#types-of-workloads)
+  - [Custom Workloads](#custom-workloads)
+  - [Parameters](#parameters)
+  - [Output vs. WorkloadResultsOutput](#output-vs-workloadresultsoutput)
+- [Suites](#suites)
+- [Spark-Submit-Configs](#spark-submit-configs)
+- [Levels and Combinations of Parallelism](#levels-and-combinations-of-parallelism)
+  - [Classic Benchmarking](#classic-benchmarking)
+  - [Classic Benchmarking Across Systems](#classic-benchmarking-across-systems)
+  - [Same Algorithm, Different Spark Settings](#same-algorithm-different-spark-settings)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Summary
+
+`spark-bench` can do two main things:
+1. Run workloads in a highly configurable fashion
+2. Generate data
+
+When running workloads in `spark-bench`, you can launch one or many `spark-submit-config`s that will contain one or many `suites`
```

Comment:
When running workloads in `spark-bench`, you can launch one or many `spark-submit-config`s that can contain one or many `suites`, which are logical groupings of one or many `workloads`.